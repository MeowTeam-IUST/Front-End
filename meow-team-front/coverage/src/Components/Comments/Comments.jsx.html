
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/Components/Comments/Comments.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/Components/Comments</a> Comments.jsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">89.88% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>80/89</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">83.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>5/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">37.5% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/8</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">89.88% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>80/89</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import styles  from '../../Pages/ProductPage/ProductPage.module.scss'
import { useState,useEffect } from 'react';
&nbsp;
export function Comment({logo, name, date, text}) {
    return (
        
        &lt;div className={styles.cm1}&gt;
          
            &lt;div className={styles.cmp1}&gt;
            &lt;div className={styles.nd}&gt;
            &lt;div className={styles.nl}&gt;
            &lt;div className={styles.logoo}&gt; {logo}&lt;/div&gt;
            &lt;div className={styles.names}&gt;{name}&lt;/div&gt;
            &lt;/div&gt;
            &lt;div className={styles.date}&gt;{date}&lt;/div&gt;
            &lt;/div&gt;
            &lt;div className={styles.rep}&gt;
                &lt;text className={styles.repicon}&gt;
                    پاسخ
                &lt;/text&gt;
                &lt;svg xmlns="http://www.w3.org/2000/svg" width="14" height="13" viewBox="0 0 14 13" fill="none"&gt;
                &lt;path d="M0.227189 4.64639L5.0398 0.490549C5.46106 0.126745 6.125 0.422089 6.125 0.987213V3.17617C10.5172 3.22646 14 4.10675 14 8.26917C14 9.9492 12.9177 11.6136 11.7214 12.4837C11.348 12.7553 10.816 12.4145 10.9536 11.9743C12.1935 8.00914 10.3655 6.95645 6.125 6.89541V9.29935C6.125 9.86537 5.46056 10.1594 5.0398 9.79601L0.227189 5.63975C-0.0755195 5.37829 -0.0759395 4.90822 0.227189 4.64639Z" fill="#E52A49"/&gt;
                &lt;/svg&gt;
            &lt;/div&gt;
            &lt;/div&gt;
            &lt;div className={styles.cmtxt}&gt;{text}&lt;/div&gt;
            
        &lt;/div&gt;
    )
}
&nbsp;
export function UserBox({userlogo, usertext, onCommentSubmit}) {
    const [text, setText] = useState('');
&nbsp;
    const [rating, setRating] = useState(0);
&nbsp;
    const handleRating = <span class="fstat-no" title="function not covered" >(rate) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        setRating(rate);</span>
<span class="cstat-no" title="statement not covered" >    };</span>
&nbsp;
    const handleSubmit = <span class="fstat-no" title="function not covered" >() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        onCommentSubmit(text, rating);</span>
<span class="cstat-no" title="statement not covered" >        setText('');</span>
<span class="cstat-no" title="statement not covered" >        setRating(0);</span>
<span class="cstat-no" title="statement not covered" >    };</span>
&nbsp;
    return (
        &lt;div className={styles.userbox}&gt;
            &lt;div className={styles.userlogo}&gt;{userlogo}&lt;/div&gt;
            &lt;input 
                className={styles.usertext} 
                value={text} 
                onChange={<span class="fstat-no" title="function not covered" >(e) =&gt; setText(e.target.value)} </span>
                placeholder={usertext}
            /&gt;
            &lt;div className={styles.sendbox}&gt;
                
            &lt;button className={styles.btsend} onClick={handleSubmit}&gt; ار سال&lt;/button&gt;
            &lt;div&gt;
                {[1, 2, 3, 4, 5].map((star) =&gt; (
                    &lt;span key={star}  className={styles.st} onClick={<span class="fstat-no" title="function not covered" >() =&gt; handleRating(star)}&gt;</span>
                        {star &lt;= <span class="branch-0 cbranch-no" title="branch not covered" >rating ? '★' : '</span>☆'}
                    &lt;/span&gt;
                ))}
            &lt;/div&gt;
            
                        &lt;/div&gt;
        &lt;/div&gt;
    )
}
&nbsp;
export function CommentsSection({comments: initialComments, userbox}) {
    const [comments, setComments] = useState(initialComments);
&nbsp;
    const handleCommentSubmit = <span class="fstat-no" title="function not covered" >(text) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        // Add the new comment to the beginning of the comments array</span>
<span class="cstat-no" title="statement not covered" >        setComments([{text, name: 'User', date: 'Just now'}, ...comments]);</span>
<span class="cstat-no" title="statement not covered" >    };</span>
&nbsp;
    return (
        &lt;div className={styles.cm}&gt;
            {comments.map((comment, index) =&gt; 
                &lt;Comment key={index} {...comment} /&gt;
            )}
            &lt;UserBox {...userbox} onCommentSubmit={handleCommentSubmit} /&gt;
        &lt;/div&gt;
    )
}</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-11-16T17:03:22.064Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    