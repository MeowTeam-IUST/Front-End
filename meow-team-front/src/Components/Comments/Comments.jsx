import React from 'react'
import styles  from '../../Pages/ProductPage/ProductPage.module.scss'
import { useState,useEffect } from 'react';

export function Comment({logo, name, date, text}) {
    return (
        
        <div className={styles.cm1}>
          
            <div className={styles.cmp1}>
            <div className={styles.nd}>
            <div className={styles.nl}>
            <div className={styles.logoo}> {logo}</div>
            <div className={styles.names}>{name}</div>
            </div>
            <div className={styles.date}>{date}</div>
            </div>
            <div className={styles.rep}>
                <text className={styles.repicon}>
                    پاسخ
                </text>
                <svg xmlns="http://www.w3.org/2000/svg" width="14" height="13" viewBox="0 0 14 13" fill="none">
                <path d="M0.227189 4.64639L5.0398 0.490549C5.46106 0.126745 6.125 0.422089 6.125 0.987213V3.17617C10.5172 3.22646 14 4.10675 14 8.26917C14 9.9492 12.9177 11.6136 11.7214 12.4837C11.348 12.7553 10.816 12.4145 10.9536 11.9743C12.1935 8.00914 10.3655 6.95645 6.125 6.89541V9.29935C6.125 9.86537 5.46056 10.1594 5.0398 9.79601L0.227189 5.63975C-0.0755195 5.37829 -0.0759395 4.90822 0.227189 4.64639Z" fill="#E52A49"/>
                </svg>
            </div>
            </div>
            <div className={styles.cmtxt}>{text}</div>
            
        </div>
    )
}

export function UserBox({userlogo, usertext, onCommentSubmit}) {
    const [text, setText] = useState('');

    const [rating, setRating] = useState(0);

    const handleRating = (rate) => {
        setRating(rate);
    };

    const handleSubmit = () => {
        onCommentSubmit(text, rating);
        setText('');
        setRating(0);
    };

    return (
        <div className={styles.userbox}>
            <div className={styles.userlogo}>{userlogo}</div>
            <input 
                className={styles.usertext} 
                value={text} 
                onChange={(e) => setText(e.target.value)} 
                placeholder={usertext}
            />
            <div className={styles.sendbox}>
                
            <button className={styles.btsend} onClick={handleSubmit}> ارسال</button>
            <div>
                {[1, 2, 3, 4, 5].map((star) => (
                    <span key={star}  className={styles.st} onClick={() => handleRating(star)}>
                        {star <= rating ? '★' : '☆'}
                    </span>
                ))}
            </div>
            
                        </div>
        </div>
    )
}

export function CommentsSection({comments: initialComments, userbox}) {
    const [comments, setComments] = useState(initialComments);

    const handleCommentSubmit = (text) => {
        // Add the new comment to the beginning of the comments array
        setComments([{text, name: 'User', date: 'Just now'}, ...comments]);
    };

    return (
        <div className={styles.cm}>
            {comments.map((comment, index) => 
                <Comment key={index} {...comment} />
            )}
            <UserBox {...userbox} onCommentSubmit={handleCommentSubmit} />
        </div>
    )
}