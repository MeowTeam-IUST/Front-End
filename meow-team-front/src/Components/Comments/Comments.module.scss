@import "../../App.scss";

.AllComments{
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  align-items: center;
  gap: 48px;
  width: 100%;
}

.CommentsList {
    display: flex;
    flex-direction: column;
    width: 100%;
    gap: 16px;
}

.Comment{
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 16px;
  flex-grow: 1;
  width: 100%;
  padding-bottom: 16px;
  border-bottom: 1px solid $--Opex--gray;
  
  .CommentHeaderSection{
    display: flex;
    justify-content: space-between;
    align-items: center;
    align-self: stretch;

    .CommentHeaderProfileAndDate{
        display: flex;
        width: 455px;
        justify-content: flex-start;
        align-items: center;
        gap: 20px;

        .CommentHeaderProfile{
            display: flex;
            justify-content: flex-end;
            align-items: center;
            gap: 10px;

            .CommentProfilePic{
                width: 32px;
                height: 32px;
                flex-shrink: 0;
                border-radius: 32px;
                background: url("https://www.uplooder.net/img/image/85/1ddd4b3d1b4cd880b4de3fe2c8a8fac2/peakpx-(2).jpg");
                background-repeat: no-repeat;
                background-size: cover;
            }
            .CommentName {

            }
        }
        .CommentDate{
            color: $color2;
            font-size: small;
            
        }
    }
    .replyButton{
        display: flex;
        align-items: center;
        gap: 8px;
        padding: 8px 16px;
        border-radius: 100px;
        border: none;

        .ReplyText{
            color: $--Opex--red;
            font-weight: bold;
            
        }
    }

    .replyButton:hover {
        background-color: $--Opex--gray;
        cursor: pointer;
    }
  }
  .CommentText{
    text-align: right;
    font-family: $second_font;

    
    direction: rtl;
  }
}




.SendSection{
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  gap: 24px;
  width: 100%;

  .userlogo{
    width: 40px;
    height: 40px;
    flex-shrink: 0;
    border-radius: 42px;
    background: url("https://www.uplooder.net/img/image/85/1ddd4b3d1b4cd880b4de3fe2c8a8fac2/peakpx-(2).jpg");
    background-repeat: no-repeat;
    background-size: cover;
   }

   .InputText{
        display: flex;
        height: 100px;
        padding: 16px 16px;
        flex-direction: column;
        justify-content: flex-start;
        flex-grow: 1;
        border-radius: 16px;
        border: none;   
    }

     .SendButtonAndRateSection{
        display: flex;
        flex-direction: column;
        align-items: baseline;
        gap: 24px;

        .SendButton{
            width: 100px;
            height: 45px;
            background-color: $--Opex--red;
            color: white;
            border: none;
            border-radius: 12px;
            font-size: medium;
            transition-duration: 0.4s;
        }

        .SendButton:hover{
            cursor: pointer;
            transform: scale(1.05);
            transition-duration: 0.4s;
        }
        .star{
            font-size: 25px;
            font-weight: 800;
            color: yellow;
            cursor: pointer;
        }
  }
   
}