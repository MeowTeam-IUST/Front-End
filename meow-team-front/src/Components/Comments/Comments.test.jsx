import { test } from "vitest";
import { Comment, UserBox, CommentsSection } from "./Comments"; // Update with the correct file path
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

test("Comment component renders correctly", () => {
  const { asFragment } = render(
    <Comment logo="logo" name="John" date="2023-01-01" text="Test comment" />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("UserBox component renders correctly", () => {
  const { asFragment } = render(
    <UserBox userlogo="user-logo" usertext="Enter your comment" />
  );
  expect(asFragment()).toMatchSnapshot();
});

// test("UserBox component handles input change correctly", () => {
//   const { getByRole } = render(
//     <UserBox userlogo="user-logo" usertext="Enter your comment" />
//   );
//   const input = getByRole("textbox");

//   userEvent.type(input, "New comment");

//   expect(input).toHaveValue("New comment");
// });

test("CommentsSection component renders correctly", () => {
  const comments = [
    { logo: "logo1", name: "John", date: "2023-01-01", text: "Comment 1" },
    { logo: "logo2", name: "Alice", date: "2023-01-02", text: "Comment 2" },
  ];

  const userbox = { userlogo: "user-logo", usertext: "Enter your comment" };

  const { asFragment } = render(
    <CommentsSection comments={comments} userbox={userbox} />
  );
  expect(asFragment()).toMatchSnapshot();
});
