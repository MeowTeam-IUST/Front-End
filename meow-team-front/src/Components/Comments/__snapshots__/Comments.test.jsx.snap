// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comment component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_cm1_f5868a"
  >
    <div
      class="_cmp1_f5868a"
    >
      <div
        class="_nd_f5868a"
      >
        <div
          class="_nl_f5868a"
        >
          <div
            class="_logoo_f5868a"
          >
             logo
          </div>
          <div
            class="_names_f5868a"
          >
            John
          </div>
        </div>
        <div
          class="_date_f5868a"
        >
          2023-01-01
        </div>
      </div>
      <div
        class="_rep_f5868a"
      >
        <text
          class="_repicon_f5868a"
        >
          پاسخ
        </text>
        <svg
          fill="none"
          height="13"
          viewBox="0 0 14 13"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.227189 4.64639L5.0398 0.490549C5.46106 0.126745 6.125 0.422089 6.125 0.987213V3.17617C10.5172 3.22646 14 4.10675 14 8.26917C14 9.9492 12.9177 11.6136 11.7214 12.4837C11.348 12.7553 10.816 12.4145 10.9536 11.9743C12.1935 8.00914 10.3655 6.95645 6.125 6.89541V9.29935C6.125 9.86537 5.46056 10.1594 5.0398 9.79601L0.227189 5.63975C-0.0755195 5.37829 -0.0759395 4.90822 0.227189 4.64639Z"
            fill="#E52A49"
          />
        </svg>
      </div>
    </div>
    <div
      class="_cmtxt_f5868a"
    >
      Test comment
    </div>
  </div>
</DocumentFragment>
`;

exports[`CommentsSection component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_cm_f5868a"
  >
    <div
      class="_cm1_f5868a"
    >
      <div
        class="_cmp1_f5868a"
      >
        <div
          class="_nd_f5868a"
        >
          <div
            class="_nl_f5868a"
          >
            <div
              class="_logoo_f5868a"
            >
               logo1
            </div>
            <div
              class="_names_f5868a"
            >
              John
            </div>
          </div>
          <div
            class="_date_f5868a"
          >
            2023-01-01
          </div>
        </div>
        <div
          class="_rep_f5868a"
        >
          <text
            class="_repicon_f5868a"
          >
            پاسخ
          </text>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 14 13"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0.227189 4.64639L5.0398 0.490549C5.46106 0.126745 6.125 0.422089 6.125 0.987213V3.17617C10.5172 3.22646 14 4.10675 14 8.26917C14 9.9492 12.9177 11.6136 11.7214 12.4837C11.348 12.7553 10.816 12.4145 10.9536 11.9743C12.1935 8.00914 10.3655 6.95645 6.125 6.89541V9.29935C6.125 9.86537 5.46056 10.1594 5.0398 9.79601L0.227189 5.63975C-0.0755195 5.37829 -0.0759395 4.90822 0.227189 4.64639Z"
              fill="#E52A49"
            />
          </svg>
        </div>
      </div>
      <div
        class="_cmtxt_f5868a"
      >
        Comment 1
      </div>
    </div>
    <div
      class="_cm1_f5868a"
    >
      <div
        class="_cmp1_f5868a"
      >
        <div
          class="_nd_f5868a"
        >
          <div
            class="_nl_f5868a"
          >
            <div
              class="_logoo_f5868a"
            >
               logo2
            </div>
            <div
              class="_names_f5868a"
            >
              Alice
            </div>
          </div>
          <div
            class="_date_f5868a"
          >
            2023-01-02
          </div>
        </div>
        <div
          class="_rep_f5868a"
        >
          <text
            class="_repicon_f5868a"
          >
            پاسخ
          </text>
          <svg
            fill="none"
            height="13"
            viewBox="0 0 14 13"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0.227189 4.64639L5.0398 0.490549C5.46106 0.126745 6.125 0.422089 6.125 0.987213V3.17617C10.5172 3.22646 14 4.10675 14 8.26917C14 9.9492 12.9177 11.6136 11.7214 12.4837C11.348 12.7553 10.816 12.4145 10.9536 11.9743C12.1935 8.00914 10.3655 6.95645 6.125 6.89541V9.29935C6.125 9.86537 5.46056 10.1594 5.0398 9.79601L0.227189 5.63975C-0.0755195 5.37829 -0.0759395 4.90822 0.227189 4.64639Z"
              fill="#E52A49"
            />
          </svg>
        </div>
      </div>
      <div
        class="_cmtxt_f5868a"
      >
        Comment 2
      </div>
    </div>
    <div
      class="_userbox_f5868a"
    >
      <div
        class="_userlogo_f5868a"
      >
        user-logo
      </div>
      <input
        class="_usertext_f5868a"
        placeholder="Enter your comment"
        value=""
      />
      <div
        class="_sendbox_f5868a"
      >
        <button
          class="_btsend_f5868a"
        >
           ار سال
        </button>
        <div>
          <span
            class="_st_f5868a"
          >
            ☆
          </span>
          <span
            class="_st_f5868a"
          >
            ☆
          </span>
          <span
            class="_st_f5868a"
          >
            ☆
          </span>
          <span
            class="_st_f5868a"
          >
            ☆
          </span>
          <span
            class="_st_f5868a"
          >
            ☆
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserBox component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_userbox_f5868a"
  >
    <div
      class="_userlogo_f5868a"
    >
      user-logo
    </div>
    <input
      class="_usertext_f5868a"
      placeholder="Enter your comment"
      value=""
    />
    <div
      class="_sendbox_f5868a"
    >
      <button
        class="_btsend_f5868a"
      >
         ار سال
      </button>
      <div>
        <span
          class="_st_f5868a"
        >
          ☆
        </span>
        <span
          class="_st_f5868a"
        >
          ☆
        </span>
        <span
          class="_st_f5868a"
        >
          ☆
        </span>
        <span
          class="_st_f5868a"
        >
          ☆
        </span>
        <span
          class="_st_f5868a"
        >
          ☆
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
